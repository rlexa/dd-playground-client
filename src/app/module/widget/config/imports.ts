import { CommonModule } from '@angular/common';
import { MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';

export const imports = [CommonModule, RouterModule, MatCardModule, MatListModule];
