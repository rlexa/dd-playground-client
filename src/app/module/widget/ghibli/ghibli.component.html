<div ddFlexbox="hyscsw">
  <mat-card class="childhv">
    <mat-card-title>Queries</mat-card-title>
    <mat-card-content>
      <div ddFlexbox="vys">
        <mat-list class="mat-list-auto-height">
          <h3 mat-subheader>API</h3>
          <mat-list-item *ngFor="let type of TYPES_LISTS">
            <button mat-raised-button (click)="toAnyData(TYPE_TO_GET_LIST[type])">{{ type }}</button>
          </mat-list-item>
        </mat-list>
      </div>
    </mat-card-content>
  </mat-card>

  <mat-card *ngIf="tableAllData$ | async as tableAllData" class="childhv">
    <mat-card-title>List</mat-card-title>
    <mat-card-content>
      <app-loading-bar [show]="(busyCount$ | async) > 0"></app-loading-bar>
      <app-simple-table
        [style.maxWidth]="'100rem'"
        [data]="tableAllData"
        [isClickable]="clickableValue"
        (clicked)="onClickedValue($event.key, $event.value)"
      ></app-simple-table>
    </mat-card-content>
  </mat-card>

  <mat-card *ngIf="anyData$ | async as anyData" class="childhv">
    <mat-card-title>Details</mat-card-title>
    <mat-card-content class="scroll-y-auto" [style.maxHeight]="'60rem'">
      <app-loading-bar [show]="(busyCount$ | async) > 0"></app-loading-bar>
      <app-simple-view
        [data]="anyData"
        [isWidthConstrained]="false"
        [isDense]="true"
        [isClickable]="clickableValue"
        (clicked)="onClickedValue($event.key, $event.value)"
      ></app-simple-view>
    </mat-card-content>
  </mat-card>
</div>
