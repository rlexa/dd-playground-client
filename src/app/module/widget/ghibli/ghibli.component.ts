import {ChangeDetectionStrategy, Component, OnDestroy, OnInit} from '@angular/core';
import {
  GhibliApiService,
  GHIBLI_TYPES,
  GHIBLI_TYPE_FILM,
  GHIBLI_TYPE_LOCATION,
  GHIBLI_TYPE_PEOPLE,
  GHIBLI_TYPE_SPECIES,
  GHIBLI_TYPE_VEHICLES,
} from 'app/module/service/ghibli-api';
import {DoneSubject, RxCleanup, rxNext_} from 'dd-rxjs';
import {BehaviorSubject, Observable, of, Subject} from 'rxjs';
import {filter, finalize, map, switchMap, tap} from 'rxjs/operators';
import {isArray} from 'util';

@Component({
  selector: 'app-ghibli',
  templateUrl: './ghibli.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class GhibliComponent implements OnDestroy, OnInit {
  constructor(private readonly api: GhibliApiService) {}

  private readonly TYPE_TO_GET_DETAILS = {
    [GHIBLI_TYPE_FILM]: this.api.film$,
    [GHIBLI_TYPE_LOCATION]: this.api.location$,
    [GHIBLI_TYPE_PEOPLE]: this.api.person$,
    [GHIBLI_TYPE_SPECIES]: this.api.species$,
    [GHIBLI_TYPE_VEHICLES]: this.api.vehicle$,
  };
  @RxCleanup() private readonly done$ = new DoneSubject();

  readonly TYPE_TO_GET_LIST = {
    [GHIBLI_TYPE_FILM]: this.api.films$,
    [GHIBLI_TYPE_LOCATION]: this.api.locations$,
    [GHIBLI_TYPE_PEOPLE]: this.api.persons$,
    [GHIBLI_TYPE_SPECIES]: this.api.speciess$,
    [GHIBLI_TYPE_VEHICLES]: this.api.vehicles$,
  };
  readonly TYPES_LISTS = Object.keys(this.TYPE_TO_GET_LIST);

  @RxCleanup() readonly anyData$ = new Subject();
  @RxCleanup() readonly busyCount$ = new BehaviorSubject(0);

  readonly tableAllData$ = this.anyData$.pipe(map(_ => (Array.isArray(_) ? _ : null)));

  ngOnDestroy() {}

  ngOnInit() {}

  clickableValue = (key: string, value: any) => typeof value === 'string' && GHIBLI_TYPES.findIndex(_ => value.includes(_)) >= 0;

  onClickedValue = (key: string, value: any) => {
    if (typeof value === 'string') {
      const ghibliType = GHIBLI_TYPES.find(_ => value.includes(_));
      if (ghibliType) {
        const id = value.substr(ghibliType.length);
        const getter = (id ? this.TYPE_TO_GET_DETAILS : this.TYPE_TO_GET_LIST)[ghibliType];
        this.toAnyData(getter, id);
      }
    }
  };

  toAnyData = <T>(switchTo: (val: T) => Observable<any>, param?: T) =>
    of(param)
      .pipe(
        filter(_ => !!switchTo),
        tap(() => this.busyCount$.next(this.busyCount$.value + 1)),
        switchMap(switchTo),
        finalize(() => this.busyCount$.next(this.busyCount$.value - 1)),
      )
      .subscribe(rxNext_(this.anyData$), rxNext_(this.anyData$));
}
