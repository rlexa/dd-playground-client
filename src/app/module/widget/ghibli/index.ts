export * from './ghibli.component';
export * from './ghibli.module';
