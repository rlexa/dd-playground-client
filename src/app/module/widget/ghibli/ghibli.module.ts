import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { ROUTE_ROOT, ROUTE_WILDCARD } from 'app/routing';
import { GhibliComponent } from './ghibli.component';
import { imports } from './imports';

const ROUTING = <Routes>[
  { path: ROUTE_ROOT, component: GhibliComponent },
  { path: ROUTE_WILDCARD, redirectTo: ROUTE_ROOT, pathMatch: 'full' },
];

@NgModule({ imports: [...imports, RouterModule.forChild(ROUTING)], exports: [GhibliComponent], declarations: [GhibliComponent] })
export class GhibliModule { }
