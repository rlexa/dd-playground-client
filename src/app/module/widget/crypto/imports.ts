import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import {
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
} from '@angular/material';
import { RouterModule } from '@angular/router';
import { CryptoApiModule } from 'app/module/service/crypto-api';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'app/module/directive/flexbox';

export const imports = [
  CommonModule,
  FormsModule,
  RouterModule,
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
  FlexboxModule,
  CryptoApiModule,
  SimpleViewModule,
];
