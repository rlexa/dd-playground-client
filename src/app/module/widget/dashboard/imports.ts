import { CommonModule } from '@angular/common';
import { MatButtonModule, MatToolbarModule, MatTooltipModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { IconPipeModule } from 'app/module/pipe/icon';
import { RoutedContentModule } from 'app/module/widget/routed-content';
import { VersionModule } from 'app/module/widget/version';
import { WidgetsModule } from 'app/widgets/widgets.module';
import { FlexboxModule } from 'dd-flexbox';
import 'hammerjs';

export const imports = [
  CommonModule, RouterModule,
  MatButtonModule, MatToolbarModule, MatTooltipModule, FlexboxModule,
  IconPipeModule, VersionModule,
  // remove
  RoutedContentModule, WidgetsModule,
];
