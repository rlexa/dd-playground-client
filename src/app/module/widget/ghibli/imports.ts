import { CommonModule } from '@angular/common';
import { MatButtonModule, MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { GhibliApiModule } from 'app/module/service/ghibli-api';
import { LoadingBarModule } from 'app/module/widget/loading-bar';
import { SimpleTableModule } from 'app/module/widget/simple-table';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'dd-flexbox';

export const imports = [
  CommonModule, RouterModule,
  MatButtonModule, MatCardModule, MatListModule, FlexboxModule,
  GhibliApiModule, LoadingBarModule, SimpleTableModule, SimpleViewModule,
];
