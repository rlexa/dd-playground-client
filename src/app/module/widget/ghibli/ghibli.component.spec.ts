import { async, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { GhibliComponent } from './ghibli.component';
import { imports } from './imports';

describe('GhibliComponent', () => {
  beforeEach(async(() => {
    TestBed.configureCompiler(<any>{ preserveWhitespaces: false }).configureTestingModule({
      imports: [NoopAnimationsModule, RouterTestingModule, ...imports], declarations: [GhibliComponent], providers: [],
    }).compileComponents();
  }));

  test('is created', () => {
    const fixture = TestBed.createComponent(GhibliComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
