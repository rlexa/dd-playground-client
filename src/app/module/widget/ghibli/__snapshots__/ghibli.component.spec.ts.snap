// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GhibliComponent is created 1`] = `
<app-ghibli
  TYPES_LISTS={[Function Array]}
  TYPE_TO_GET_DETAILS={[Function Object]}
  TYPE_TO_GET_LIST={[Function Object]}
  anyData$={[Function Subject]}
  api={[Function GhibliApiService]}
  busyCount$={[Function BehaviorSubject]}
  clickableValue={[Function Function]}
  done$={[Function DoneSubject]}
  onClickedValue={[Function Function]}
  tableAllData$={[Function AnonymousSubject]}
  toAnyData={[Function Function]}
>
  <div
    ddflexbox="hyscsw"
    ng-reflect-dd-flexbox="hyscsw"
    style="display: flex; flex-direction: row; justify-content: flex-start; align-items: flex-start; align-content: flex-start; flex-wrap: wrap;"
  >
    <mat-card
      class="childhv mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        Queries
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <div
          ddflexbox="vys"
          ng-reflect-dd-flexbox="vys"
          style="display: flex; flex-direction: column; justify-content: flex-start; align-items: flex-start; align-content: stretch; flex-wrap: no-wrap;"
        >
          <mat-list
            class="mat-list-auto-height mat-list"
          >
            <h3
              class="mat-subheader"
              mat-subheader=""
            >
              API
            </h3>
            
            <mat-list-item
              class="mat-list-item"
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="true"
                  ng-reflect-trigger="[object HTMLUnknownElement]"
                />
                <div
                  class="mat-list-text"
                />
                <button
                  class="mat-raised-button _mat-animation-noopable"
                  mat-raised-button=""
                >
                  <span
                    class="mat-button-wrapper"
                  >
                    /films/
                  </span>
                  <div
                    class="mat-button-ripple mat-ripple"
                    matripple=""
                    ng-reflect-centered="false"
                    ng-reflect-disabled="false"
                    ng-reflect-trigger="[object HTMLButtonElement]"
                  />
                  <div
                    class="mat-button-focus-overlay"
                  />
                </button>
              </div>
            </mat-list-item>
            <mat-list-item
              class="mat-list-item"
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="true"
                  ng-reflect-trigger="[object HTMLUnknownElement]"
                />
                <div
                  class="mat-list-text"
                />
                <button
                  class="mat-raised-button _mat-animation-noopable"
                  mat-raised-button=""
                >
                  <span
                    class="mat-button-wrapper"
                  >
                    /locations/
                  </span>
                  <div
                    class="mat-button-ripple mat-ripple"
                    matripple=""
                    ng-reflect-centered="false"
                    ng-reflect-disabled="false"
                    ng-reflect-trigger="[object HTMLButtonElement]"
                  />
                  <div
                    class="mat-button-focus-overlay"
                  />
                </button>
              </div>
            </mat-list-item>
            <mat-list-item
              class="mat-list-item"
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="true"
                  ng-reflect-trigger="[object HTMLUnknownElement]"
                />
                <div
                  class="mat-list-text"
                />
                <button
                  class="mat-raised-button _mat-animation-noopable"
                  mat-raised-button=""
                >
                  <span
                    class="mat-button-wrapper"
                  >
                    /people/
                  </span>
                  <div
                    class="mat-button-ripple mat-ripple"
                    matripple=""
                    ng-reflect-centered="false"
                    ng-reflect-disabled="false"
                    ng-reflect-trigger="[object HTMLButtonElement]"
                  />
                  <div
                    class="mat-button-focus-overlay"
                  />
                </button>
              </div>
            </mat-list-item>
            <mat-list-item
              class="mat-list-item"
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="true"
                  ng-reflect-trigger="[object HTMLUnknownElement]"
                />
                <div
                  class="mat-list-text"
                />
                <button
                  class="mat-raised-button _mat-animation-noopable"
                  mat-raised-button=""
                >
                  <span
                    class="mat-button-wrapper"
                  >
                    /species/
                  </span>
                  <div
                    class="mat-button-ripple mat-ripple"
                    matripple=""
                    ng-reflect-centered="false"
                    ng-reflect-disabled="false"
                    ng-reflect-trigger="[object HTMLButtonElement]"
                  />
                  <div
                    class="mat-button-focus-overlay"
                  />
                </button>
              </div>
            </mat-list-item>
            <mat-list-item
              class="mat-list-item"
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="true"
                  ng-reflect-trigger="[object HTMLUnknownElement]"
                />
                <div
                  class="mat-list-text"
                />
                <button
                  class="mat-raised-button _mat-animation-noopable"
                  mat-raised-button=""
                >
                  <span
                    class="mat-button-wrapper"
                  >
                    /vehicles/
                  </span>
                  <div
                    class="mat-button-ripple mat-ripple"
                    matripple=""
                    ng-reflect-centered="false"
                    ng-reflect-disabled="false"
                    ng-reflect-trigger="[object HTMLButtonElement]"
                  />
                  <div
                    class="mat-button-focus-overlay"
                  />
                </button>
              </div>
            </mat-list-item>
          </mat-list>
        </div>
      </mat-card-content>
    </mat-card>
    
    
  </div>
</app-ghibli>
`;
